src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int MAX_NUMBER_OF_FIGURE = 4;
+    public static final int MAX_SIZE_OF_PARAMETER = 10;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final Circle circle = new Circle("white", 10);
```

Comment:
Don't create it here, you will always return a reference to one single object. Instead, move the magic values to constants `DEFAULT_NUMBER` and `DEFAULT_COLOR`.